### PR TITLE
Preprocess update

### LINKED ISSUE #3: Make the phylogeny optional
If not specified, then we can skip UniFrac metrics.

### LINKED ISSUE #6: Produce rarefied and un-rarefied table
From the preprocess pipeline.

### PR DESCRIPTION
**Pull request by patrickimran**
Addresses #3 and #6. 
If no phylogeny is given, we return empty distance matrices for the phylogenetic outputs. 
Preprocess now produces two output tables: a rarefied table, and an unrarefied table whose samples are filtered to match those in the rarefied table.

### CODE REVIEW

File: q2_mlab/tests/test_preprocessing.py [hunk 1]
```diff
@@ -51,17 +52,49 @@ def test_preprocess_output(self):
...
+        self.assertEqual(len(results), 9)
+        self.assertTrue(str(results[0].type) == 'FeatureTable[Frequency]')
+        self.assertTrue(str(results[1].type) == 'FeatureTable[Frequency]')
+        self.assertTrue(str(results[2].type) == 'SampleData[Target]')
+        self.assertTrue(str(results[3].type) == 'DistanceMatrix')
+        self.assertTrue(str(results[4].type) == 'DistanceMatrix')
+        self.assertTrue(str(results[5].type) == 'DistanceMatrix')
+        self.assertTrue(str(results[6].type) == 'DistanceMatrix')
+        self.assertTrue(str(results[7].type) == 'DistanceMatrix')
+        self.assertTrue(str(results[8].type) == 'DistanceMatrix')
```
Comment (ElDeveloper) [inline q2_mlab/tests/test_preprocessing.py:93]: Can you load these two objects and check they have a shape 0, 0?

**Review by ElDeveloper — CHANGES_REQUESTED** [review]
Looks great, just one minor comment.

File: q2_mlab/tests/test_preprocessing.py (hunk 1 above)
Comment (patrickimran) [reply q2_mlab/tests/test_preprocessing.py:93]: Added in a check. When I create the "empty" distance matrices, a (1,1) matrix is the smallest I can go:
```
In [9]: dm = DistanceMatrix(data=[])                                                                                                                                                                                           

In [10]: dm.data                                                                                                                                                                                                               
Out[10]: array([[0.]])
```
The saved weighted_unifrac_distance_matrix.qza and unweighted_unifrac_distance_matrix.qza will all be 28kb, which is noticeably smaller than the others. Not able to rename the output files just for this condition AFAIK.

**Comment by ElDeveloper** [comment]
Thanks @patrickimran

**Pull request merged**